src/mito2/src/sst/parquet/row_group.rs

```diff
@@ -166,19 +184,28 @@ impl<'a> InMemoryRowGroup<'a> {
                 .map(|(idx, (_chunk, _cached_pages))| {
                     let column = self.metadata.column(idx);
                     let (start, length) = column.byte_range();
-                    start as usize..(start + length) as usize
+                    start..(start + length)
                 })
-                .collect();
+                .collect::<Vec<_>>();
 
             if fetch_ranges.is_empty() {
                 // Nothing to fetch.
                 return Ok(());
             }
 
+            // Try to fetch data from WriteCache,
+            // If not in WriteCache, fetch data from object store.
+            let key = (self.region_id, self.file_id);
             let mut chunk_data =
-                fetch_byte_ranges(self.file_path, self.object_store.clone(), fetch_ranges)
-                    .await?
-                    .into_iter();
+                if let Some(data) = self.fetch_ranges_from_write_cache(key, &fetch_ranges).await {
+                    data.into_iter()
+                } else {
+                    // Fetch data from object store directly.
+                    fetch_byte_ranges(self.file_path, self.object_store.clone(), &fetch_ranges)
+                        .await
+                        .map_err(|e| ParquetError::External(Box::new(e)))?
+                        .into_iter()
+                };
```

Comment:
Consider wrapping a function or struct to do this and eliminating some duplicated codes.